src/Exception/HttpExceptionTrait.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Superbrave\VerboseErrorHttpClient\Exception;
+
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Throwable;
+
+/**
+ * @author Niels Nijens <nn@superbrave.nl>
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+trait HttpExceptionTrait
+{
+    /**
+     * @var ResponseInterface
+     */
+    private $response;
+
+    /**
+     * @param ResponseInterface $response
+     * @param Throwable         $previousException
+     */
+    public function __construct(ResponseInterface $response, Throwable $previousException)
+    {
+        $this->response = $response;
+
+        $code = $response->getInfo('http_code');
+        $url = $response->getInfo('url');
+        $message = sprintf('HTTP %d returned for "%s".', $code, $url);
+
+        $httpCodeFound = false;
+        $isJson = false;
+        foreach (array_reverse($response->getInfo('response_headers')) as $h) {
+            if (0 === strpos($h, 'HTTP/')) {
+                if ($httpCodeFound) {
+                    break;
+                }
+
+                $message = sprintf('%s returned for "%s".', $h, $url);
+                $httpCodeFound = true;
+            }
+
+            if (0 === stripos($h, 'content-type:')) {
+                if (preg_match('/\bjson\b/i', $h)) {
+                    $isJson = true;
+                }
+
+                if ($httpCodeFound) {
+                    break;
+                }
+            }
+        }
+
+        $contentSummary = $this->getResponseContentSummary($isJson);
+        if (empty($contentSummary) === false) {
```

Comment:
empty will never return anything other than a boolean, so why not just use `if (!empty($contentSummary)) { .. }` ?

Reply:
The exclamation mark is quickly overlooked, so I think this is more readable.